Observer/CreateOrder.php

```diff
@@ -58,6 +59,10 @@ public function execute(\Magento\Framework\Event\Observer $observer)
             $order->save();
             $this->_monduLogger->logTransaction($order, $orderData, null);
         } catch (Exception $e) {
+            $orderPayment = $order->getPayment();
+            $orderPayment->deny(false);
+            $order->setStatus(Order::STATE_CANCELED);
```

Comment:
I guess you intentionally still kept it yeah? even though we solved the original issue

Reply:
Yea just in case something happens but now we shouldn't get here at all.